x-pack/test/api_integration/apis/search/search.ts

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function ({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const esArchiver = getService('esArchiver');
+
+  describe('search', () => {
+    describe('post', () => {
+      it('should return 200 with final response if wait_for_completion_timeout is long enough', async () => {
+        const resp = await supertest
+          .post(`/internal/search/ese`)
+          .set('kbn-xsrf', 'foo')
+          .send({
+            params: {
+              body: {
+                query: {
+                  match_all: {},
+                },
+              },
+              wait_for_completion_timeout: '1000s',
+            },
+          })
+          .expect(200);
+
+        const { id } = resp.body;
+        expect(id).to.be(undefined);
+        expect(resp.body.isPartial).to.be(false);
+        expect(resp.body.isRunning).to.be(false);
+        expect(resp.body).to.have.property('rawResponse');
+      });
+
+      it('should return 200 with partial response if wait_for_completion_timeout is long enough', async () => {
+        const resp = await supertest
+          .post(`/internal/search/ese`)
+          .set('kbn-xsrf', 'foo')
+          .send({
+            params: {
+              body: {
+                query: {
+                  match_all: {},
+                },
+              },
+              wait_for_completion_timeout: '1ms',
+            },
+          })
+          .expect(200);
+
+        const { id } = resp.body;
+        expect(id).not.to.be(undefined);
+        expect(resp.body.isPartial).to.be(true);
+        expect(resp.body.isRunning).to.be(true);
+        expect(resp.body).to.have.property('rawResponse');
+      });
+
+      it('should retrieve results with id', async () => {
+        const resp = await supertest
+          .post(`/internal/search/ese`)
+          .set('kbn-xsrf', 'foo')
+          .send({
+            params: {
+              body: {
+                query: {
+                  match_all: {},
+                },
+              },
+              wait_for_completion_timeout: '1ms',
+            },
+          })
+          .expect(200);
+
+        const { id } = resp.body;
+
+        await new Promise((resolve) => setTimeout(resolve, 2000));
```

Comment:
Is there any reason we wouldn't just use `wait_for_completion_timeout` in this 2nd request?

Reply:
Yes, because the GET request doesn't support passing in an override for `wait_for_completion_timeout`